spec/models/show_spec.rb

```diff
@@ -1,5 +1,29 @@
+# frozen_string_literal: true
+
 require 'rails_helper'
 
 RSpec.describe Show, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  let(:show) { create(:show) }
+  let(:title) { show.title }
+  let(:show_type) { show.show_type }
+
+  it "is a valid with the presence of show_type, show_id and title" do
+    expect(show).to be_valid
+  end
+
+  context "when title is not present" do
+    let(:title) { nil }
+    
+    it "is invalid" do
+      expect(subject).to be_invalid
+    end
+  end
+
+  context "when show_type is not 'Movie' or 'TV Show' " do
+    let(:show_type) { 'Short' }
```

Comment:
this is not really doing anything, you have to set these variables on the object... what you can do is create your own subject like:
```
subject {
    described_class.new(title: "Anything",
                        show_type: "Type")
  }
```
and then in context block do something like:
```
context "when show_type is not 'Movie' or 'TV Show' " do
      it "is invalid" do
          subject.title = nil
           expect(subject).to be_invalid
    end 
end
```
or just use the shoulda matchers gem.

Reply:
the thing is, if you try and do something like:
```
let(:show_type) { "TV Show" }
```
This spec will still says that `expect(subject).to be_invalid` is true.

Reply:
meaning this show_type variable is not assigned anywhere

Reply:
That makes sense, I ran it with the `binding.pry` and you are right. I will change it following your suggestions :) thanks Goran!